### DISCUSSION TITLE
Outreachy Discussions - 2022 Round 2

### DISCUSSION THREAD

**Discussion by MridulS** [networkx/networkx#6016] [General]
Hello all!

Please use this discussion page to discuss everything for Outreachy 2022 round 2 (December 2022 - March 2023).

If you have any questions or doubts about the processes please put in a reply to this discussion post.

As a first step: applicants who haven't used networkx before should first [install networkx](https://networkx.org/documentation/latest/install.html) on their machines and follow our [introductory tutorial](https://networkx.org/documentation/latest/tutorial.html). This will give applicants a better overview of our codebase and API, which will help the applicant create a strong application. If something isn't clear from the tutorial please raise an issue/put a comment here so we can do a better job of maintaining our tutorial.

Once applicants are comfortable with using networkx, they should go through our [Contributor Guide](https://networkx.org/documentation/latest/developer/contribute.html) to setup our local development environment for networkx.

We have created another repository https://github.com/networkx/outreachy which will have explicit tasks for Outreachy applicants during the contribution period.

To participate with NetworkX as an Outreachy intern applicants are expected to go through multiple tasks in that repository. All tasks are presented as [issues](https://github.com/networkx/outreachy/issues). We will add a couple of more tasks to this repository.

Everyone is expected to start with https://github.com/networkx/outreachy/issues/2.

The applicant should also go through the [New Contributor FAQs](https://networkx.org/documentation/latest/developer/new_contributor_faq.html) as it may answer some of their questions.

As the project will deal with creating new notebooks for our NX-guides please do go through current notebooks at https://networkx.org/nx-guides/.

If the applicant is interested in working on specific issues they can comment on them. Applicants can also pick any [open issue](https://github.com/networkx/networkx/issues) if they like or even open a new issue/discussion if there is something in our documentation or codebase that doesn't make sense :)

Feel free to post here if you have any other questions or issues!

Excited for the coming winter! :)

**Comment by nsengiyumva-wilberforce** [comment]
Hi!, I am interested in contributing to this project for my outreachy contributions, what steps do I need to follow to be accepted here?

**Reply by MridulS** [reply]
Hello @nsengiyumva-wilberforce! Thanks for your interest.

As a first step: applicants who haven't used networkx before should first [install networkx](https://networkx.org/documentation/latest/install.html) on their machines and follow our [introductory tutorial](https://networkx.org/documentation/latest/tutorial.html). This will give applicants a better overview of our codebase and API, which will help the applicant create a strong application. If something isn't clear from the tutorial please raise an issue/put a comment here so we can do a better job of maintaining our tutorial.

**Reply by nsengiyumva-wilberforce** [reply]
sure, thanks your reply, i have finished the installation. now reading through the tutorial. what editor or @MridulS, what editor or IDE do you recommend me to use for this project. currently, I am using visual studio for the tutrial practice!

**Reply by MridulS** [reply]
@nsengiyumva-wilberforce there isn't any IDE recommendation from me :) Feel free to use any which you are comfortable with.
Visual studio should work just fine.

**Reply by nsengiyumva-wilberforce** [reply]
Ok, thanks.

**Comment by abrhamgg** [comment]
Hello, i am an Outreachy applicant , i am interested to start contributing,
 any guidelines?

**Reply by nsengiyumva-wilberforce** [reply]
hello @abrhamgg, set up [install networkx](https://networkx.org/documentation/latest/install.html) on your local machine and work through[introductory tutorial](https://networkx.org/documentation/latest/tutorial.html).

**Comment by abrhamgg** [comment]
Hello @MridulS, i have finished the first task and i have created a pull request.

**Comment by Georgiedata** [comment]
Hello, I am Emesowum George Chidozie and I am interested in contributing to NetworkX. I have gone through the description and I would like to know if it is okay to install NetworkX on Jupyter Notebook...

**Reply by gladensP** [reply]
Yes, you can install it on Jupyter Notebook. First make sure that you have the latest version of pip installed and that your python version is 3.8 or higher.

**Reply by AgnesWangechi** [reply]
How can I install through Jupyter Notebook? I am currently stuck at the installation part.

**Reply by chimaobi-okite** [reply]
inside a python cell, run "!pip install networkx[default]" then restart the kernel

**Reply by AgnesWangechi** [reply]
Thank you it was installed successfully. However, I am trying to run the tutorial but i don't seem to see any output on the notebook.

**Reply by chimaobi-okite** [reply]
Can you explain better? Like if you run print('hello-world) on your notebook, you don't get any results?

**Reply by chimaobi-okite** [reply]
> Can you explain better? Like if you run print('hello-world) on your notebook, you don't get any results?

if yes, check [this](https://github.com/ipython/ipython/issues/11027)

**Comment by gladensP** [comment]
Hello @MridulS , I am Gladens Popoola and I am an Outreachy applicant. I am interested in contributing to Networkx. I have gone through the descriptions and instructions and I have completed the first task and I have made a pull request.

**Comment by RoseZhen2022** [comment]
Hello everyone. I'm Rose, an Outreachy applicant interested in contributing to Networkx. This is the first time I use Networks. By going through the tutorials I found this is a powerful tool to study graphs and networks. I have complete the first task and created a pull request. @MridulS I look forward to learning more from this project.

**Reply by nsengiyumva-wilberforce** [reply]
@RoseZhen2022 you are very welcome, I have just finished it. It's a very interesting project. I loved drawing graphs bit of he tutorials. Yeah the tutorial helped me understand more the algorithms I studied at the college

**Comment by alaminumar** [comment]
Hello Everyone . My name is Alamin , I am an outreachy applicant and i am very much interested in the project .

**Reply by nsengiyumva-wilberforce** [reply]
Hi @alaminumar , welcome here. please [install networkx](https://networkx.org/documentation/latest/install.html) on your machine and read through the [introductory tutorial](https://networkx.org/documentation/latest/tutorial.html). And do https://github.com/networkx/outreachy/issues/2

**Comment by kusanele** [comment]
Good day everyone. My name is Kusanele .Its good to be here. Looking forward to contributing to this project with you all.

**Comment by chimaobi-okite** [comment]
Hello everyone, I'm chimaobi. Looking forward to learning a lot while contributing to this project. It's nice meeting you all.

**Reply by nsengiyumva-wilberforce** [reply]
Hi, welcome here, nice to meet you too

**Comment by Hauwarh** [comment]
Hi everyone
My name is Hauwa.
Looking forward to contributing to this project.

**Reply by nsengiyumva-wilberforce** [reply]
Welcome, feel free here, we have a very nice mentor

**Reply by Hauwarh** [reply]
Thank you

**Comment by nsengiyumva-wilberforce** [comment]
@MridulS , are we supposed to continue with another  or we should wait for you to assign us? I am asking on behalf of those whose PRs have been accepted

**Reply by MridulS** [reply]
Please go ahead and start working on other issues at https://github.com/networkx/outreachy/issues. No need to assign, there are general tasks so everyone can do them :)

**Reply by nsengiyumva-wilberforce** [reply]
ok, thanks

**Comment by phayena** [comment]
Hi everyone  my name is mariam fayena, nice to meet you all i hope to learn from the program

**Reply by kusanele** [reply]
Hey @phayena , Welcome to the party

**Comment by kusanele** [comment]
@MridulS How many interns will be  selected for this project??

**Reply by MridulS** [reply]
We are aiming for 1 right now.

**Reply by kusanele** [reply]
Thanks for a prompt response , more pressure in the house 🥴

**Reply by nsengiyumva-wilberforce** [reply]
😂 haha

**Reply by kusanele** [reply]
@nsengiyumva-wilberforce I think they should Atleast select 2

**Reply by nsengiyumva-wilberforce** [reply]
You never know, anyway, they might select more than one.

**Reply by kusanele** [reply]
Fingures crossed

**Reply by Zzee2** [reply]
even though it's just one person, the lessons that will be taken from here are going to be valuable!

**Comment by Zzee2** [comment]
Hi everyone, am so glad to be here, wen/t through the tutorials and and am glad to be here

**Comment by tindi-plus** [comment]
Hello Everyone, I am Tindi Sommers. I am looking forward to working with you all.

**Reply by Zzee2** [reply]
Hello Tindi!, welcome

**Reply by tindi-plus** [reply]
Thank you Zzee2

**Comment by Georgiedata** [comment]
Nice to meet you, Tindi

**Reply by tindi-plus** [reply]
Thank you Georgiedata

**Comment by Georgiedata** [comment]
![Screenshot (189)](https://user-images.githubusercontent.com/102357428/194970980-abe9d92b-1993-4f23-bc43-e61c8f86526d.png)
 I am having challenges setting up by build environment using Anaconda prompt .  From the contributor guide, I copied the codes and tried installing through the prompt.

**Reply by nsengiyumva-wilberforce** [reply]
Hi make sure you forked and cloned the right project and then cd into the project root to run that command ou want to run

**Reply by chimaobi-okite** [reply]
Hi, @Georgiedata,  you're creating your virtual environment in C:\Users\HP and that is not where you cloned your networkx repo to. The repo should be in something like  C:\Users\HP\*\networkx (* can be one or more subfolders). Find the right folder, cd to it, create the virtual environment inside it, and run your commands, everything should work fine.

**Reply by Georgiedata** [reply]
Ok, and this should be run on Anaconda Prompt, correct?

**Reply by chimaobi-okite** [reply]
yes. Say your networkx folder is in C:\Users\HP\Documents\networkx. then :-
cd Documents\networkx - takes you to the project root directory (C:\Users\HP\Documents\networkx>).
conda create --name networkx-dev - creates a conda environment with the name networkx-dev inside the project. Activate it then continue as specified in the documentation

**Reply by Zzee2** [reply]
If any of the mentioned solutions didn't work, follow this, I used it to get my conda running properly before running the installation again

https://stackoverflow.com/questions/66066600/anaconda-install-error-conda-script-py-h-v-command

**Reply by Georgiedata** [reply]
Thanks @chimaobi-okite for the feedback and others. I have found the path and activated networkx , but trying the next line of code _'conda install -c conda-forge --file requirements/default.txt --file requirements/test.txt --file requirements/developer.txt' brings up file directory issues again. Prior to this project, I had been using Jupyter through Anaconda, I don't know if I would have to deactivate the previous enironment...
'

**Reply by Zzee2** [reply]
Do you run your jupyter on terminal before going to anaconda?, if yes, run it on terminal instead of going to anaconda before installing, else I will suggest, just install the package as a whole on your local machine without conda path, it makes it alot easier for you and it will work perfectly fine on jupyter notebook

**Reply by chimaobi-okite** [reply]
> Thanks @chimaobi-okite for the feedback and others. I have found the path and activated networkx , but trying the next line of code _'conda install -c conda-forge --file requirements/default.txt --file requirements/test.txt --file requirements/developer.txt' brings up file directory issues again. Prior to this project, I had been using Jupyter through Anaconda, I don't know if I would have to deactivate the previous enironment... '

can you post a screenshot of your anaconda prompt showing the commands and error?

**Comment by nsengiyumva-wilberforce** [comment]
@MridulS , can you please recommend for some issue to start with?

thanks
Wilberforce

**Reply by MridulS** [reply]
@nsengiyumva-wilberforce we have added some good first issues which improve our test coverage here https://github.com/networkx/networkx/issues?q=is%3Aissue+is%3Aopen+label%3A%22Good+First+Issue%22

**Reply by nsengiyumva-wilberforce** [reply]
Thank you so much

**Comment by phayena** [comment]
In case you are getting the 
**CondaSSLError: OpenSSL appears to be unavailable on this machine. OpenSSL is required to
download and install packages.**  error message after upgrading your conda
just downgrade it back to be able to download the packages use 
**conda install conda==4.14.0**
or read here   [https://github.com/conda/conda/issues/11795](url)

Hope this helps someone

**Comment by ladykkk** [comment]
Hi, everyone.
My name is Jiafei Kuang.
Nice to meet you all here and hope to make contribute to this project.:)

**Reply by nsengiyumva-wilberforce** [reply]
You are very welcome, feel free, we have a very very nice mentor

**Comment by raianegoncalves** [comment]
can anybody help me? I'm having trouble pulling the repository to visual studio code with git clone, every time I try to give git clone git@github.com:raianegoncalves/outreachy.git, the error message appears: git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

It says I'm permission denied. How to proceed?

**Reply by nsengiyumva-wilberforce** [reply]
Did you setup github ssh key? if not do so! follow [this](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)

**Comment by nsengiyumva-wilberforce** [comment]
@MridulS, I will be glad to have a good resource that I can read for writing tests, recommend please?

**Reply by MridulS** [reply]
We use [pytest](https://docs.pytest.org/en/7.1.x/) to write tests. In my opinion the best way to write tests would be to read our current test suite. Just read through files which start with `test_` in the networkx codebase and you can find examples and strategies to write tests.

**Reply by nsengiyumva-wilberforce** [reply]
noted!Thanks for the advice

**Comment by jane-ai98** [comment]
Hello everyone, I'm Jane Harry, I'm looking forward to contributing to this project. I'm currently learning about Networkx, and network science. I'm excited about gaining new knowledge and skills through this experience. I'll always be in the discussion session, as I assume I'd have lots of questions moving forward. thank you for this opportunity.

**Reply by nsengiyumva-wilberforce** [reply]
You are very welcome @jane-ai98 , feel free here. We have a very nice mentor @MridulS

**Comment by chimaobi-okite** [comment]
For anyone having issues with checks while raising PRs, try to use a Linker(flake8) - it helps ensure your code conforms to the coding style. Then after you push, check your actions tab to be sure all tests ran successfully, if it does, you can then go ahead and raise your PR. 

I had issues with this yesterday especially the style check, hope this helps someone too

**Reply by nsengiyumva-wilberforce** [reply]
I added it but it is still failing

**Reply by chimaobi-okite** [reply]
Maybe allow black to automatically do the formatting for you. If you're on VS code , follow [this](https://marcobelo.medium.com/setting-up-python-black-on-visual-studio-code-5318eba4cd00) steps. You can investigate GitHub actions to see which checks failed and why when you push, that way you can easily debug

**Comment by nsengiyumva-wilberforce** [comment]
@MridulS , this https://github.com/networkx/networkx/pull/6066 is failing one test. I need help

**Comment by jane-ai98** [comment]
Hello mentor... I'm stuck at the learning phase. I have installed networkx on my system, and I'm learning about the API and its documentation. Are there any other materials I can consult to get better information, how do I proceed from this stage.

**Reply by nsengiyumva-wilberforce** [reply]
Hi!, the mentor is @MridulS , where are you stuck, post anything that is not coming out, we can help.
A documentation could be a better place to start from. but what i always do is to google around, read and work through some simple examples and then keep the documentation my guard.

Wilberforce

**Comment by tindi-plus** [comment]
Hello @MridulS, I have completed the first task and have created a pull request. However, I am not seeing my folder in ''2022-round-2''.  My username is tindi-plus.

**Reply by MridulS** [reply]
I have merged the pull request. You should see the https://github.com/networkx/outreachy/tree/main/2022-round-2/tindi-plus folder now.

**Reply by tindi-plus** [reply]
Thank you.

**Comment by RoseZhen2022** [comment]
Hi @MridulS , is there any possible to add more Good first issues ? It seems that all of them have been taken. I have completed all other four tasks, what should I do next?

**Reply by MridulS** [reply]
@RoseZhen2022 you can go ahead and go through the coverage report of networkx https://app.codecov.io/gh/networkx/networkx/tree/main/networkx/algorithms and look at files which doesn't have 100% test coverage. There are a lot of places you can add new tests to make our test suite more comprehensive :)

Avoid the isomorphism folder there.

**Comment by paulitapb** [comment]
Hi! I'm Paula! I'm so excited to contribute here! If anyone needs some help installing networkX or using git I can help! feel free to ask!

**Reply by nsengiyumva-wilberforce** [reply]
Hi @paulitapb , you are welcome

**Comment by Fati0381** [comment]
Hi Everyone my name is Fatima, I am interested in contributing to Networkx, am sorry for joining you late, I was suffering from chicken fox, but thank God am feeling better now, hope to work with you guys, thank you.

**Reply by nsengiyumva-wilberforce** [reply]
Hi @Fati0381 , we hope you be well. Welcome here

**Reply by Fati0381** [reply]
Hope so, Thank you.

**Comment by dnabanita7** [comment]
Hello, I am an outreachy applicant. I don't find any open Good first issues but I see most of them concentrate on test coverage. Can I open work on something from the tests?

**Reply by MridulS** [reply]
Sure! Please feel free to work on any other open issue. Please do go through the issues available at https://github.com/networkx/outreachy first.

**Reply by chimaobi-okite** [reply]
Hi @dnabanita7 , welcome. 
I think your question have been addressed [here](https://github.com/networkx/networkx/discussions/6016#discussioncomment-3883864)

**Reply by dnabanita7** [reply]
Thank you! The introductory tutorial is pretty well written.

**Comment by AnneO1** [comment]
Hi everyone,
Am Anne, an Outreachy applicant starting on NetworkX. I hope to collaborate and learn from you all.

**Reply by chimaobi-okite** [reply]
Hi, @AnneO1. Welcome 💐

**Comment by nsengiyumva-wilberforce** [comment]
@MridulS , I made some changes in test_line.py. you can have a look at my functions and may comment on them. Currently when I generate a coverage report, it's at 95%. I hope to make it 100% soon. I am enjoying this because i had never used pytest before but I am getting a great experience

Thanks
Wilberforce

**Comment by rahmah18** [comment]
Hello guys... I'm Rahmah, hoping to learn from you all

**Reply by nsengiyumva-wilberforce** [reply]
Hello @rahmah18 , welcome here. You are gonna have a good time here with our mentor @MridulS , He is so grate

**Comment by nsengiyumva-wilberforce** [comment]
@MridulS I am have a hard time with testing line 478 in test_line.py. Could you consider looking at it and may be advise me? All the test cases I am trying are always true!!

**Reply by nsengiyumva-wilberforce** [reply]
It's the only one remaining and would be glad to finish ASAP, will appreciate your help

**Comment by tindi-plus** [comment]
Hello Team,
A quick question: When I created a PR in NetworkX repository, I got this notification on the description page 
"
Please run black to format your code.
See https://networkx.org/documentation/latest/developer/contribute.html for details.
"
I remember during the environment setup for NetworkX, we were asked to install pre-commit which is supposed to run **black** before our code can be committed. 
My question is: what exactly is black?
I understand that pre-commit is a hook that is supposed to run some code before committing your changes. ( I Googled it). I know that **black** is one of those hooks. But what does it do?

**Reply by dschult** [reply]
Black is a code styling tool. It has a fairly strict and sensible way to arrange line-breaks, indentation, blank lines, etc. Some people would call it a `linter`. The reason to use it is that everybody is then following the same style choices and we can avoid wasting time on PRs talking about the style of the code. :)  At least that's my quick summary. I may be missing other features and/or reasons for using it.

**Reply by tindi-plus** [reply]
Thank you @dschult for the explanation. This makes a lot of sense.

**Comment by jeftersantiago** [comment]
Hi everyone, i'm kinda late here, but i'm very excited to contribute with NetworkX and learn from you all!

**Reply by chimaobi-okite** [reply]
Welcome @jeftersantiago

**Reply by tindi-plus** [reply]
Great having you here. Hope you are finding your way around.

**Comment by paulitapb** [comment]
Hi! How's everyone doing? 
@MridulS We are asked for a timeline of the project to submit our final application. Do you have a rough estimate or should we plan it?

**Reply by MridulS** [reply]
A rough timeline for the pedagogical notebooks project should follow something like
Week of Dec 5: Research about the algorithm, dataset or method for the pedagogical notebook 1
 Week of Dec 12: Implementation and Review 
Week of Dec 19: Review
Week of Dec 26: Research about the algorithm, dataset or method for the pedagogical notebook 2
 Week of Jan 2: Implementation and Review 
Week of Jan 9: Review
Week of Jan 16: Research about the algorithm, dataset or method for the pedagogical notebook 3 
Week of Jan 23: Implementation and Review 
Week of Jan 30: Review
Week of Feb 6: Research about the algorithm, dataset or method for the pedagogical notebook 4
 Week of Feb 13: Implementation and Review 
Week of Feb 20: Review
 Week of Feb 27: Wrap up of the internship
Feel free to add more or less details to the timeline.

**Reply by paulitapb** [reply]
Thanks!!

**Comment by tindi-plus** [comment]
Hello Team,

Another quick question:

I See lots of open pull request on networkx repo that appears to have no conflicts, and based on the comments (from reviewers) on the PR, the PRs are satisfactory. My question is: why are they still open? Is there a particular time when all accepted PRs are merged?

**Reply by MridulS** [reply]
Generally for a PR to be merged there needs to be 2 independent reviews. I have been overriding this for some of the PRs with just one review so outreachy contributors don't have to wait too much for getting their PR merged. I'm trying to merge all the PRs for outreachy which I think are good to go without waiting for the second review.

If there are any old PRs that has 2 approved reviews (green ticks) and not merged please let me know :)

**Reply by tindi-plus** [reply]
Okay, thank you @MridulS  for the explanation. I haven't been checking for two approved reviews though.

**Reply by nsengiyumva-wilberforce** [reply]
> Generally for a PR to be merged there needs to be 2 independent reviews. I have been overriding this for some of the PRs with just one review so outreachy contributors don't have to wait too much for getting their PR merged. I'm trying to merge all the PRs for outreachy which I think are good to go without waiting for the second review.
> 
> If there are any old PRs that has 2 approved reviews (green ticks) and not merged please let me know :)

I am wondering if we are the ones supposed to look for reviewers, @MridulS is it our job, some people just come and review?

**Reply by MridulS** [reply]
Maintainers will review open PRs when they have the time, you can request a review from specific people if you think someone specific should be reviewing the particular PR.

**Reply by nsengiyumva-wilberforce** [reply]
> Maintainers will review open PRs when they have the time, you can request a review from specific people if you think someone specific should be reviewing the particular PR.

It looks most of them are pretty busy, but we shall wait patiently. We humbly request for your help on this to be done before the application period ends.

And personally, I thank you for your reviews on my PRs, I know they have delayed there but your criticism is a learning point to me and I don't take it for granted. 

I look forward to get the 2 reviewers ASAP

**Reply by MridulS** [reply]
> We humbly request for your help on this to be done before the application period ends.

Just to make it clear, you don't *need* to have it merged before application. Just an open PR is enough :)

**Reply by nsengiyumva-wilberforce** [reply]
ok, thanks

**Comment by ladykkk** [comment]
Hi @MridulS I have two PRs opened several days ago but they are still not been reviewed. They are #6083 and #6087. The coverage of them are 100 % and 97.5% now. Can you please take a look at them? Thank you so much:)

**Comment by Qudirah** [comment]
Hello! My name is Qudirah. Took me this long to introduce myself. Happy to be here. <3

**Reply by Qudirah** [reply]
Was wondering if we would need to add our first set of good issues in the outreach repo to our contributions

**Reply by MridulS** [reply]
> Was wondering if we would need to add our first set of good issues in the outreach repo to our contributions

You just need to add your PRs to the outreachy repo.

**Reply by Qudirah** [reply]
Let me rephrase my question when I'm adding my contributions to my application, do I need to add the PRs in the outreachy repo (e.g pedagogical notebook, the nx_version file, etc.)? Do I add each of those ones as contributions too? @MridulS

**Reply by MridulS** [reply]
Yes you need to add them too as they are contributions too.

**Reply by nsengiyumva-wilberforce** [reply]
> Yes you need to add them too as they are contributions too.

ok, we shall include them too then

**Reply by Qudirah** [reply]
> Yes you need to add them too as they are contributions too.

Thank you

**Reply by Qudirah** [reply]
@MridulS please check on #6161

**Comment by tindi-plus** [comment]
I guess this is the end.

Nice time with everyone.

**Reply by nsengiyumva-wilberforce** [reply]
nice time to you,
